### PR TITLE
rustc: Remove some dead code

### PR DESCRIPTION
**Pull request by petrochenkov**
Extracted from https://github.com/rust-lang/rust/pull/43192

r? @eddyb

### CODE REVIEW

File: src/librustc_mir/dataflow/impls/mod.rs [hunk 1]
```diff
@@ -204,40 +201,6 @@ impl<'a, 'tcx: 'a> HasMoveData<'tcx> for DefinitelyInitializedLvals<'a, 'tcx> {
...
-/// operation reinitializing that l-value.
-///
-/// The resulting dataflow is a more enriched version of
-/// `MaybeUninitializedLvals`. Both structures on their own only tell
-/// you if an l-value *might* be uninitialized at a given point in the
-/// control flow. But `MovingOutStatements` also includes the added
-/// data of *which* particular statement causing the deinitialization
-/// that the borrow checker's error message may need to report.
-#[allow(dead_code)]
-pub struct MovingOutStatements<'a, 'tcx: 'a> {
```
Comment (eddyb) [inline src/librustc_mir/dataflow/impls/mod.rs]: r? @pnkfelix or @arielb1

**Comment by eddyb** [comment]
LGTM except for some MIR changes - I'm not sure if anyone has plans for that code.

File: src/librustc_mir/dataflow/impls/mod.rs (hunk 1 above)
Comment (arielb1) [inline src/librustc_mir/dataflow/impls/mod.rs]: This is supposed to be used by MIR borrowck.

Comment (pnkfelix) [inline src/librustc_mir/dataflow/impls/mod.rs]: Its okay, this is easy to put back when the time comes.

File: src/librustc_data_structures/graph/mod.rs [hunk 2]
```diff
@@ -282,25 +251,6 @@ impl<N: Debug, E: Debug> Graph<N, E> {
         self.incoming_edges(target).sources()
     }
 
-    /// A common use for graphs in our compiler is to perform
-    /// fixed-point iteration. In this case, each edge represents a
-    /// constraint, and the nodes themselves are associated with
-    /// variables or other bitsets. This method facilitates such a
-    /// computation.
-    pub fn iterate_until_fixed_point<'a, F>(&'a self, mut op: F)
```
Comment (pnkfelix) [inline src/librustc_data_structures/graph/mod.rs]: I was pretty surprised that this is dead code. But as far as I can tell, #7688 introduced it alongside with another fixed-point iteration function definition ... and I guess no one ended up ever invoking the one that used the new graph abstraction ... Maybe that was meant to be future work that never got done? Anyway, I don't object to removing it now.

Comment (pnkfelix) [inline src/librustc_data_structures/graph/mod.rs]: (oh I see; even #7688 says that the new CFG code is not yet used...)

**Pull request merged**